jolt-evm-verifier/src/subprotocols/R1CSMatrix.sol

```diff
@@ -0,0 +1,520 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.0;
+
+import {MODULUS, Fr, FrLib} from "./Fr.sol";
+import "forge-std/console.sol";
+
+/// Allows calculation of the R1CS step matrix
+library R1CSMatrix {
+    using FrLib for Fr;
+
+    /// Evaluates the z mle for the outer sumcheck
+    /// @param r The r we are evaluating on
+    /// @param segmentEvals The segment evals from the proof, should be length 89
+    function eval_z_mle(Fr[] memory r, uint256[] memory segmentEvals) internal pure returns (Fr) {
+        require(segmentEvals.length == 89, "Segment Length must be 89");
+        // We calculate the evals based on a length 7 slice
+        Fr[] memory r_var_eqs = eq_poly_evals(r, 1, 7);
```

Comment:
let's pull 89 and 7 our into named constants

Reply:
These can be named but the code may break if you change them, so I generally don't prefer named constants which can't be changed.